src/common.rs

```diff
@@ -784,13 +790,56 @@ impl PartialEq<FQNameRef<'_>> for FQName {
 pub enum Identifier {
     /// This variant is case sensitive
     /// "fOo""bAr""" is stored as fOo"bAr"
-    Quoted(String),
+    Quoted(IdentifierWithSpan),
     /// This variant is case insensitive
     /// Only ascii alphanumeric and _ characters are allowed in this variant
     /// fOo_bAr is stored as fOo_bAr
-    Unquoted(String),
+    Unquoted(IdentifierWithSpan),
+}
+
+#[derive(Debug, Clone, Eq, Ord, PartialOrd, PartialEq, Deserialize, Hash)]
+pub struct IdentifierWithSpan {
+    value: String,
+    span: Option<Span>,
+}
+
+impl From<&str> for IdentifierWithSpan {
+    fn from(value: &str) -> Self {
+        IdentifierWithSpan {
+            value: value.to_owned(),
+            span: None,
+        }
+    }
+}
+
+impl From<String> for IdentifierWithSpan {
+    fn from(value: String) -> Self {
+        IdentifierWithSpan { value, span: None }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Hash)]
+pub struct Span {
+    pub start: tree_sitter::Point,
+    pub end: tree_sitter::Point,
 }
 
+impl<'a> From<&tree_sitter::Node<'a>> for Span {
+    fn from(value: &tree_sitter::Node<'a>) -> Self {
+        Self {
+            start: value.start_position(),
+            end: value.end_position(),
+        }
+    }
+}
+impl<'de> Deserialize<'de> for Span {
```

Comment:
we can land this as is, I'll clean it up in https://github.com/shotover/rust-cql3-parser/pull/50